# Avoir les 3 sites de naturalité la plus grande et la plus petite

------

In [ ]:
import pandas as pd
import numpy as np

In [ ]:

df = pd.read_csv('naturalness_index_by_site.csv')

In [ ]:
df.head(2).T

0                  1
CODE_TRANS               375_rouge_estive  375_rouge_lisiere
LAYER                                   1                  1
mean_naturalness_100             148.0625           149.4375
std_naturalness_100             48.909954           38.85513
median_naturalness_100              185.0              154.0
mean_naturalness_250            171.03719         169.739669
std_naturalness_250             39.241447          39.661981
median_naturalness_250              185.0              185.0
mean_naturalness_500           172.802838         168.812723
std_naturalness_500             37.376364          43.425038
median_naturalness_500              185.0              185.0
mean_naturalness_1000          174.770356         178.830256
std_naturalness_1000            37.411821          41.650766
median_naturalness_1000             185.0              185.0

In [ ]:
# Supposons que ta DataFrame s'appelle df
df[['Transect', 'Dynamique', 'Position']] = df['CODE_TRANS'].str.split('_', expand=True)

In [ ]:
df

CODE_TRANS  LAYER  mean_naturalness_100  std_naturalness_100  \
0     375_rouge_estive      1            148.062500            48.909954   
1    375_rouge_lisiere      1            149.437500            38.855130   
2      374_bleu_estive      1            185.025316            13.866429   
3     374_bleu_lisiere      1            170.129870             7.685964   
4     373_rouge_estive      1            173.128205            15.538504   
..                 ...    ...                   ...                  ...   
587  454_rouge_lisiere      4            589.435897            10.803121   
588   386_rouge_estive      4            551.935065             3.971776   
589  386_rouge_lisiere      4            583.310811            26.099563   
590    228_bleu_estive      4            543.987342             7.873194   
591   228_bleu_lisiere      4            506.324675            82.710643   

     median_naturalness_100  mean_naturalness_250  std_naturalness_250  \
0                     185.0            171.037190            39.241447   
1                     154.0            169.739669            39.661981   
2                     185.0            181.563525            25.439418   
3                     169.0            175.248473            19.136964   
4                     169.0            164.320487            33.727802   
..                      ...                   ...                  ...   
587                   590.0            584.319756            24.456154   
588                   551.0            558.738866            20.068270   
589                   577.5            581.907975            28.688756   
590                   549.0            525.399594            68.339658   
591                   534.0            545.170385            81.004967   

     median_naturalness_250  mean_naturalness_500  std_naturalness_500  \
0                     185.0            172.802838            37.376364   
1                     185.0            168.812723            43.425038   
2                     185.0            179.506374            34.444274   
3                     169.0            175.295316            35.534418   
4                     169.0            172.955680            38.132923   
..                      ...                   ...                  ...   
587                   589.0            580.980642            46.923612   
588                   552.0            558.209077            60.251260   
589                   570.0            574.222336            58.086679   
590                   549.0            553.177189            69.617987   
591                   551.0            564.479857            75.933985   

     median_naturalness_500  mean_naturalness_1000  std_naturalness_1000  \
0                     185.0             174.770356             37.411821   
1                     185.0             178.830256             41.650766   
2                     185.0             183.158665             37.934693   
3                     185.0             181.411105             41.284772   
4                     185.0             167.549828             54.314766   
..                      ...                    ...                   ...   
587                   588.0             582.658816             57.708788   
588                   566.0             572.046358             66.187156   
589                   587.0             575.385911             67.061866   
590                   565.0             571.500255             68.011955   
591                   570.0             575.344204             68.837553   

     median_naturalness_1000 Transect Dynamique Position  
0                      185.0      375     rouge   estive  
1                      185.0      375     rouge  lisiere  
2                      185.0      374      bleu   estive  
3                      185.0      374      bleu  lisiere  
4                      185.0      373     rouge   estive  
..                       ...      ...       ...      .

In [ ]:
df_transect = pd.read_csv('transect_commune_ld.csv')

df_transect

Unnamed: 0  Transect       Commune                 LD
0            0        38   Formiguères  Serra dels Alarbs
1            1        44   Formiguères  Serra dels Alarbs
2            2        46  Valcebollère    Pla des Salines
3            3        47  Valcebollère    Pla des Salines
4            4       172    Les Angles         Roc d'Aude
..         ...       ...           ...                ...
69          69       638       Bestiac           Trimouns
70          70       641       Bestiac           Trimouns
71          71       646          Appy       Etang d'Appy
72          72       682          Orlu              RNCFS
73          73       702        Antras              Isard

[74 rows x 4 columns]

In [ ]:
df_transect.dtypes

Unnamed: 0     int64
Transect       int64
Commune       object
LD            object
dtype: object

In [ ]:
df.dtypes

CODE_TRANS                  object
LAYER                        int64
mean_naturalness_100       float64
std_naturalness_100        float64
median_naturalness_100     float64
mean_naturalness_250       float64
std_naturalness_250        float64
median_naturalness_250     float64
mean_naturalness_500       float64
std_naturalness_500        float64
median_naturalness_500     float64
mean_naturalness_1000      float64
std_naturalness_1000       float64
median_naturalness_1000    float64
Transect                    object
Dynamique                   object
Position                    object
dtype: object

In [ ]:
# Maintenant, change le type de la colonne 'Transect' en int64
df['Transect'] = df['Transect'].astype('int64')

In [ ]:
df = pd.merge(df, df_transect, on='Transect', how='inner')
df

CODE_TRANS  LAYER  mean_naturalness_100  std_naturalness_100  \
0     375_rouge_estive      1            148.062500            48.909954   
1    375_rouge_lisiere      1            149.437500            38.855130   
2     375_rouge_estive      2          10525.537500            14.567038   
3    375_rouge_lisiere      2          10592.387500             7.737076   
4     375_rouge_estive      3            344.425000             0.685109   
..                 ...    ...                   ...                  ...   
587   228_bleu_lisiere      2           9857.103896            10.049985   
588    228_bleu_estive      3            337.392405             1.417948   
589   228_bleu_lisiere      3            340.844156             1.414810   
590    228_bleu_estive      4            543.987342             7.873194   
591   228_bleu_lisiere      4            506.324675            82.710643   

     median_naturalness_100  mean_naturalness_250  std_naturalness_250  \
0                     185.0            171.037190            39.241447   
1                     154.0            169.739669            39.661981   
2                   10525.5          10525.314050            34.758799   
3                   10592.5          10592.357438            19.022451   
4                     345.0            344.904959             0.770891   
..                      ...                   ...                  ...   
587                  9857.0           9857.405680            22.317589   
588                   337.0            338.032454             3.055763   
589                   341.0            341.734280             3.607573   
590                   549.0            525.399594            68.339658   
591                   534.0            545.170385            81.004967   

     median_naturalness_250  mean_naturalness_500  std_naturalness_500  \
0                     185.0            172.802838            37.376364   
1                     185.0            168.812723            43.425038   
2                   10525.0          10520.001521            63.336105   
3                   10592.0          10583.039695            41.892890   
4                     345.0            345.423213             1.261865   
..                      ...                   ...                  ...   
587                  9857.0           9858.772565            43.659410   
588                   339.0            340.691446             5.406256   
589                   341.0            343.122896             5.543458   
590                   549.0            553.177189            69.617987   
591                   551.0            564.479857            75.933985   

     median_naturalness_500  mean_naturalness_1000  std_naturalness_1000  \
0                     185.0             174.770356             37.411821   
1                     185.0             178.830256             41.650766   
2                   10526.0           10500.769084            117.086845   
3                   10580.0           10557.227938             91.170465   
4                     345.0             346.934860              2.592675   
..                      ...                    ...                   ...   
587                  9858.0            9864.636815             99.810409   
588                   341.0             342.820137              7.607837   
589                   344.0             344.352739              7.392889   
590                   565.0             571.500255             68.011955   
591                   570.0             575.344204             68.837553   

     median_naturalness_1000  Transect Dynamique Position  Unnamed: 0  \
0                      185.0       375     rouge   estive          27   
1                      185.0       375     rouge  lisiere          27   
2                    10514.0       375     rouge   estive          27   
3                    10549.0       375     rouge  lisiere          27   
4                      346.0       375     

-----

## LAYER = 4 :



In [ ]:
df_layer_4 = df[df['LAYER'] == 4].copy()

### Rayon = 100 :

In [ ]:

df_layer_4_sorted_descending_100 = df_layer_4.sort_values(by='mean_naturalness_100', ascending=False)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_descending_100 = df_layer_4_sorted_descending_100.iloc[:3]

df_layer_4_sorted_ascending_100 = df_layer_4.sort_values(by='mean_naturalness_100', ascending=True)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_ascending_100 = df_layer_4_sorted_ascending_100.iloc[:3]

# Affichage des résultats
print("Trois premières lignes avec les plus grands mean_naturalness_100 :\n", df_layer_4_selected_rows_descending_100)
print("Trois premières lignes avec les plus petits mean_naturalness_100 :\n", df_layer_4_selected_rows_ascending_100)

Trois premières lignes avec les plus grands mean_naturalness_100 :
             CODE_TRANS  LAYER  mean_naturalness_100  std_naturalness_100  \
175  413_rouge_lisiere      4            634.636364            19.078041   
279   428_bleu_lisiere      4            634.050633            57.318099   
183   424_bleu_lisiere      4            627.822785            26.790684   

     median_naturalness_100  mean_naturalness_250  std_naturalness_250  \
175                   620.0            620.307536            36.719184   
279                   654.0            641.431211            43.612665   
183                   623.0            636.336735            31.752404   

     median_naturalness_250  mean_naturalness_500  std_naturalness_500  \
175                   619.0            620.209077            40.118020   
279                   636.0            639.527013            44.794610   
183                   625.0            636.110714            37.633443   

     median_naturalness_500  mean

* Les sites avec la plus grande naturalité :
- **564** : Couflens Saubé
- **265** : Couflens Pouill
----
* Les sites avec la plus faible naturalité :
- **228** : Saint-lary Col de l'Herbe Soulette
- **172** : Les Angeles Roc d'Aude
- **474** : Goulier Val de Sos

In [ ]:
df_layer_4_selected_rows_descending_100[['Annee','Transect', 'Commune', 'LD', 'Dynamique', 'Position', 'mean_naturalness_100', 'std_naturalness_100']]

KeyError: "['Annee'] not in index"

In [ ]:
df_layer_4_selected_rows_descending_100[['Transect', 'Commune', 'LD', 'Dynamique', 'Position', 'mean_naturalness_100', 'std_naturalness_100']]

Transect   Commune            LD Dynamique Position  \
175       413      Seix  Col de Pause     rouge  lisiere   
279       428  Couflens        Pouill      bleu  lisiere   
183       424      Seix  Col de Pause      bleu  lisiere   

     mean_naturalness_100  std_naturalness_100  
175            634.636364            19.078041  
279            634.050633            57.318099  
183            627.822785            26.790684

In [ ]:
df_layer_4_selected_rows_ascending_100[['Transect', 'Commune', 'LD', 'Dynamique', 'Position', 'mean_naturalness_100', 'std_naturalness_100']]

Transect     Commune          LD Dynamique Position  \
519       172  Les Angles  Roc d'Aude      bleu  lisiere   
119       474     Goulier  Val de Sos      bleu  lisiere   
518       172  Les Angles  Roc d'Aude      bleu   estive   

     mean_naturalness_100  std_naturalness_100  
519            482.367089            72.010227  
119            495.797468            45.968999  
518            498.137500            70.955222

### Rayon = 250:

In [ ]:

df_layer_4_sorted_descending_250 = df_layer_4.sort_values(by='mean_naturalness_250', ascending=False)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_descending_250 = df_layer_4_sorted_descending_250.iloc[:3]

df_layer_4_sorted_ascending_250 = df_layer_4.sort_values(by='mean_naturalness_250', ascending=True)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_ascending_250 = df_layer_4_sorted_ascending_250.iloc[:3]

# Affichage des résultats
print("Trois premières lignes avec les plus grands mean_naturalness_250 :\n", df_layer_4_selected_rows_descending_250)
print("Trois premières lignes avec les plus petits mean_naturalness_250 :\n", df_layer_4_selected_rows_ascending_250)

Trois premières lignes avec les plus grands mean_naturalness_250 :
            CODE_TRANS  LAYER  mean_naturalness_100  std_naturalness_100  \
279  428_bleu_lisiere      4            634.050633            57.318099   
183  424_bleu_lisiere      4            627.822785            26.790684   
215  401_bleu_lisiere      4            621.062500            56.460239   

     median_naturalness_100  mean_naturalness_250  std_naturalness_250  \
279                   654.0            641.431211            43.612665   
183                   623.0            636.336735            31.752404   
215                   654.0            633.715746            43.188167   

     median_naturalness_250  mean_naturalness_500  std_naturalness_500  \
279                   636.0            639.527013            44.794610   
183                   625.0            636.110714            37.633443   
215                   653.0            631.598577            39.577436   

     median_naturalness_500  mean_nat

In [ ]:
df_layer_4_selected_rows_descending_250[['Transect', 'Commune', 'LD','Dynamique', 'Position', 'mean_naturalness_250', 'std_naturalness_250']]

Transect   Commune            LD Dynamique Position  \
279       428  Couflens        Pouill      bleu  lisiere   
183       424      Seix  Col de Pause      bleu  lisiere   
215       401     Bonac          Orle      bleu  lisiere   

     mean_naturalness_250  std_naturalness_250  
279            641.431211            43.612665  
183            636.336735            31.752404  
215            633.715746            43.188167

In [ ]:
df_layer_4_selected_rows_ascending_250[['Transect', 'Commune', 'LD', 'Dynamique', 'Position','mean_naturalness_250', 'std_naturalness_250']]

Transect     Commune          LD Dynamique Position  \
254       641     Bestiac    Trimouns     rouge   estive   
519       172  Les Angles  Roc d'Aude      bleu  lisiere   
119       474     Goulier  Val de Sos      bleu  lisiere   

     mean_naturalness_250  std_naturalness_250  
254            500.133874            71.355596  
519            502.715746            67.487142  
119            505.270325            61.352226

### Rayon = 500:

In [ ]:

df_layer_4_sorted_descending_500 = df_layer_4.sort_values(by='mean_naturalness_500', ascending=False)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_descending_500 = df_layer_4_sorted_descending_500.iloc[:3]

df_layer_4_sorted_ascending_500 = df_layer_4.sort_values(by='mean_naturalness_500', ascending=True)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_ascending_500 = df_layer_4_sorted_ascending_500.iloc[:3]

# Affichage des résultats
print("Trois premières lignes avec les plus grands mean_naturalness_500 :\n", df_layer_4_selected_rows_descending_500)
print("Trois premières lignes avec les plus petits mean_naturalness_500 :\n", df_layer_4_selected_rows_ascending_500)

Trois premières lignes avec les plus grands mean_naturalness_500 :
            CODE_TRANS  LAYER  mean_naturalness_100  std_naturalness_100  \
279  428_bleu_lisiere      4            634.050633            57.318099   
183  424_bleu_lisiere      4            627.822785            26.790684   
182   424_bleu_estive      4            605.087500            11.837645   

     median_naturalness_100  mean_naturalness_250  std_naturalness_250  \
279                   654.0            641.431211            43.612665   
183                   623.0            636.336735            31.752404   
182                   606.0            621.905350            24.767962   

     median_naturalness_250  mean_naturalness_500  std_naturalness_500  \
279                   636.0            639.527013            44.794610   
183                   625.0            636.110714            37.633443   
182                   614.0            631.726305            40.665836   

     median_naturalness_500  mean_nat

In [ ]:
df_layer_4_selected_rows_descending_500[['Transect', 'Commune', 'LD','Dynamique', 'Position', 'mean_naturalness_500', 'std_naturalness_500']]

Transect   Commune            LD Dynamique Position  \
279       428  Couflens        Pouill      bleu  lisiere   
183       424      Seix  Col de Pause      bleu  lisiere   
182       424      Seix  Col de Pause      bleu   estive   

     mean_naturalness_500  std_naturalness_500  
279            639.527013            44.794610  
183            636.110714            37.633443  
182            631.726305            40.665836

In [ ]:
df_layer_4_selected_rows_ascending_500[['Transect', 'Commune', 'LD', 'Dynamique', 'Position','mean_naturalness_500', 'std_naturalness_500']]

Transect     Commune          LD Dynamique Position  \
511       200  Les Angles  Roc d'Aude     rouge  lisiere   
254       641     Bestiac    Trimouns     rouge   estive   
255       641     Bestiac    Trimouns     rouge  lisiere   

     mean_naturalness_500  std_naturalness_500  
511            503.845761            65.179886  
254            504.444501            67.758132  
255            504.611933            69.244851

### Rayon = 1000:

In [ ]:

df_layer_4_sorted_descending_1000 = df_layer_4.sort_values(by='mean_naturalness_1000', ascending=False)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_descending_1000 = df_layer_4_sorted_descending_1000.iloc[:3]

df_layer_4_sorted_ascending_1000 = df_layer_4.sort_values(by='mean_naturalness_1000', ascending=True)

# Sélection des trois premières lignes correspondantes
df_layer_4_selected_rows_ascending_1000 = df_layer_4_sorted_ascending_1000.iloc[:3]

# Affichage des résultats
print("Trois premières lignes avec les plus grands mean_naturalness_1000 :\n", df_layer_4_selected_rows_descending_1000)
print("Trois premières lignes avec les plus petits mean_naturalness_1000 :\n", df_layer_4_selected_rows_ascending_1000)

Trois premières lignes avec les plus grands mean_naturalness_1000 :
            CODE_TRANS  LAYER  mean_naturalness_100  std_naturalness_100  \
278   428_bleu_estive      4            617.532468            12.016854   
279  428_bleu_lisiere      4            634.050633            57.318099   
183  424_bleu_lisiere      4            627.822785            26.790684   

     median_naturalness_100  mean_naturalness_250  std_naturalness_250  \
278                   618.0            614.645621            28.386564   
279                   654.0            641.431211            43.612665   
183                   623.0            636.336735            31.752404   

     median_naturalness_250  mean_naturalness_500  std_naturalness_500  \
278                   618.0            618.597144            37.317988   
279                   636.0            639.527013            44.794610   
183                   625.0            636.110714            37.633443   

     median_naturalness_500  mean_na

In [ ]:
df_layer_4_selected_rows_descending_1000[['Transect', 'Commune', 'LD','Dynamique', 'Position', 'mean_naturalness_1000', 'std_naturalness_1000']]

Transect   Commune            LD Dynamique Position  \
278       428  Couflens        Pouill      bleu   estive   
279       428  Couflens        Pouill      bleu  lisiere   
183       424      Seix  Col de Pause      bleu  lisiere   

     mean_naturalness_1000  std_naturalness_1000  
278             630.249809             44.202429  
279             628.459824             54.207987  
183             623.446269             52.824315

In [ ]:
df_layer_4_selected_rows_ascending_1000[['Transect', 'Commune', 'LD', 'Dynamique', 'Position','mean_naturalness_1000', 'std_naturalness_1000']]

Transect     Commune          LD Dynamique Position  \
254       641     Bestiac    Trimouns     rouge   estive   
255       641     Bestiac    Trimouns     rouge  lisiere   
519       172  Les Angles  Roc d'Aude      bleu  lisiere   

     mean_naturalness_1000  std_naturalness_1000  
254             513.602623             67.124021  
255             513.713776             68.822826  
519             519.444756             78.444384

----

In [ ]:
print('Trois sites avec naturalité la plus grande avec rayon = 100: \n ',df_layer_4_selected_rows_descending_100[['Commune', 'LD', 'mean_naturalness_100', 'std_naturalness_100']], '\n')
print('Trois sites avec naturalité la plus grande avec rayon = 250: \n ',df_layer_4_selected_rows_descending_250[['Commune', 'LD', 'mean_naturalness_250', 'std_naturalness_250']], '\n')
print('Trois sites avec naturalité la plus grande avec rayon = 500: \n ',df_layer_4_selected_rows_descending_500[['Commune', 'LD', 'mean_naturalness_500', 'std_naturalness_500']], '\n')
print('Trois sites avec naturalité la plus grande avec rayon = 1000: \n ',df_layer_4_selected_rows_descending_1000[['Commune', 'LD', 'mean_naturalness_1000', 'std_naturalness_1000']], '\n')

Trois sites avec naturalité la plus grande avec rayon = 100: 
        Commune            LD  mean_naturalness_100  std_naturalness_100
175      Seix  Col de Pause            634.636364            19.078041
279  Couflens        Pouill            634.050633            57.318099
183      Seix  Col de Pause            627.822785            26.790684 

Trois sites avec naturalité la plus grande avec rayon = 250: 
        Commune            LD  mean_naturalness_250  std_naturalness_250
279  Couflens        Pouill            641.431211            43.612665
183      Seix  Col de Pause            636.336735            31.752404
215     Bonac          Orle            633.715746            43.188167 

Trois sites avec naturalité la plus grande avec rayon = 500: 
        Commune            LD  mean_naturalness_500  std_naturalness_500
279  Couflens        Pouill            639.527013            44.794610
183      Seix  Col de Pause            636.110714            37.633443
182      Seix  Col de P

In [ ]:
print('Trois sites avec naturalité la plus grande avec rayon = 100: \n ',df_layer_4_selected_rows_ascending_100[['Commune', 'LD', 'mean_naturalness_100', 'std_naturalness_100']], '\n')
print('Trois sites avec naturalité la plus grande avec rayon = 250: \n ',df_layer_4_selected_rows_ascending_250[['Commune', 'LD', 'mean_naturalness_250', 'std_naturalness_250']], '\n')
print('Trois sites avec naturalité la plus grande avec rayon = 500: \n ',df_layer_4_selected_rows_ascending_500[['Commune', 'LD', 'mean_naturalness_500', 'std_naturalness_500']], '\n')
print('Trois sites avec naturalité la plus grande avec rayon = 1000: \n ',df_layer_4_selected_rows_ascending_1000[['Commune', 'LD', 'mean_naturalness_1000', 'std_naturalness_1000']], '\n')

## LAYER = 2 :